### PR TITLE
Fixes the test intermediates layout for clean builds

### LINKED ISSUE #2217: Error in test build in `build clean`
Got error messages at the end of `build clean` about errors in tests.
Corresponding part of Tests_Managed_Windows_NT__x64__Debug.log:
```
Ошибка сборки. (Building error)

                   "d:\net\coreclr\tests\build.proj" ((целевой объект) target object rebuild ) (1) ->
                   "d:\net\coreclr\tests\src\dirs.proj" ((целевой объект)target object Build ) (2:2) ->
                   "d:\net\coreclr\tests\src\readytorun\mainv1.csproj" (целевой объект Build ) (4512) ->
                   (Целевой объект ResolveAssemblyReferences) -> 
                     C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(1819,5): warning MSB3245: Не удалось разрешить данную ссылку. Не удалось обнаружить сборку "test". Убедитесь, что сборка существует на диске. Если данная ссылка требуется в коде, это может вызвать ошибки компиляции. [d:\net\coreclr\tests\src\readytorun\mainv1.csproj]
                     C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(1819,5): warning MSB3243: Не удается решить конфликт между "test, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" и "test". Произвольный выбор "test, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a". [d:\net\coreclr\tests\src\readytorun\mainv1.csproj]


                   "d:\net\coreclr\tests\build.proj" (целевой объект rebuild ) (1) ->
                   "d:\net\coreclr\tests\src\dirs.proj" (целевой объект Build ) (2:2) ->
                   "d:\net\coreclr\tests\src\readytorun\mainv2.csproj" (целевой объект Build ) (4514) ->
                     C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(1819,5): warning MSB3245: Не удалось разрешить данную ссылку. Не удалось обнаружить сборку "test". Убедитесь, что сборка существует на диске. Если данная ссылка требуется в коде, это может вызвать ошибки компиляции. [d:\net\coreclr\tests\src\readytorun\mainv2.csproj]
                     C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(1819,5): warning MSB3243: Не удается решить конфликт между "test, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" и "test". Произвольный выбор "test, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a". [d:\net\coreclr\tests\src\readytorun\mainv2.csproj]


                   "d:\net\coreclr\tests\build.proj" (целевой объект rebuild ) (1) ->
                   "d:\net\coreclr\tests\src\dirs.proj" (целевой объект Build ) (2:2) ->
                   "d:\net\coreclr\tests\src\JIT\Directed\PREFIX\unaligned\1\add.ilproj" (целевой объект Build ) (4680) ->
                   (Целевой объект CoreCompile) -> 
                     EXEC : warning : Layout specified for auto-layout class [d:\net\coreclr\tests\src\JIT\Directed\PREFIX\unaligned\1\add.ilproj]


                   "d:\net\coreclr\tests\build.proj" (целевой объект rebuild ) (1) ->
                   "d:\net\coreclr\tests\src\dirs.proj" (целевой объект Build ) (2:2) ->
                   "d:\net\coreclr\tests\src\JIT\Directed\PREFIX\volatile\1\add.ilproj" (целевой объект Build ) (4715) ->
                     EXEC : warning : Layout specified for auto-layout class [d:\net\coreclr\tests\src\JIT\Directed\PREFIX\volatile\1\add.ilproj]


                   "d:\net\coreclr\tests\build.proj" (целевой объект rebuild ) (1) ->
                   "d:\net\coreclr\tests\src\dirs.proj" (целевой объект Build ) (2:2) ->
                   "d:\net\coreclr\tests\src\JIT\jit64\regress\vsw\153682\test.ilproj" (целевой объект Build ) (4852) ->
                     test.il(16): warning : Non-nested class has nested visibility (0x00000102), changed to non-nested (0x00000101) [d:\net\coreclr\tests\src\JIT\jit64\regress\vsw\153682\test.ilproj]


                   "d:\net\coreclr\tests\build.proj" (целевой объект rebuild ) (1) ->
                   "d:\net\coreclr\tests\src\dirs.proj" (целевой объект Build ) (2:2) ->
                   "d:\net\coreclr\tests\src\Interop\ReversePInvoke\Marshalling\MarshalBoolArray.csproj" (целевой объект Build ) (2665) ->
                   "d:\net\coreclr\tests\src\Interop\ReversePInvoke\Marshalling\MarshalBoolArray.csproj" (целевой объект CopyNativeProjectBinaries ) (2665:2) ->
                   (Целевой объект CopyNativeProjectBinaries) -> 
                     d:\net\coreclr\tests\src\dir.targets(112,6): error : The native project files are missing in d:\net\coreclr\tests\..\bin\tests\obj\Windows_NT.x64.Debug\src\Interop\ReversePInvoke\Marshalling\Debug\ please run build from the root of the repo at least once [d:\net\coreclr\tests\src\Interop\ReversePInvoke\Marshalling\MarshalBoolArray.csproj]


                   "d:\net\coreclr\tests\build.proj" (целевой объект rebuild ) (1) ->
                   "d:\net\coreclr\tests\src\dirs.proj" (целевой объект Build ) (2:2) ->
                   "d:\net\coreclr\tests\src\JIT\Directed\StructABI\StructABI.csproj" (целевой объект Build ) (3007) ->
                   "d:\net\coreclr\tests\src\JIT\Directed\StructABI\StructABI.csproj" (целевой объект CopyNativeProjectBinaries ) (3007:2) ->
                     d:\net\coreclr\tests\src\dir.targets(112,6): error : The native project files are missing in d:\net\coreclr\tests\..\bin\tests\obj\Windows_NT.x64.Debug\src\JIT\Directed\StructABI\Debug\ please run build from the root of the repo at least once [d:\net\coreclr\tests\src\JIT\Directed\StructABI\StructABI.csproj]


                   "d:\net\coreclr\tests\build.proj" (целевой объект rebuild ) (1) ->
                   "d:\net\coreclr\tests\src\dirs.proj" (целевой объект Build ) (2:2) ->
                   "d:\net\coreclr\tests\src\JIT\Methodical\structs\systemvbringup\structinregs.csproj" (целевой объект Build ) (3971) ->
                   "d:\net\coreclr\tests\src\JIT\Methodical\structs\systemvbringup\structinregs.csproj" (целевой объект CopyNativeProjectBinaries ) (3971:2) ->
                     d:\net\coreclr\tests\src\dir.targets(112,6): error : The native project files are missing in d:\net\coreclr\tests\..\bin\tests\obj\Windows_NT.x64.Debug\src\JIT\Methodical\structs\systemvbringup\Debug\ please run build from the root of the repo at least once [d:\net\coreclr\tests\src\JIT\Methodical\structs\systemvbringup\structinregs.csproj]


                   "d:\net\coreclr\tests\build.proj" (целевой объект rebuild ) (1) ->
                   "d:\net\coreclr\tests\src\dirs.proj" (целевой объект Build ) (2:2) ->
                   "d:\net\coreclr\tests\src\Loader\NativeLibs\FromNativePaths.csproj" (целевой объект Build ) (4506) ->
                   "d:\net\coreclr\tests\src\Loader\NativeLibs\FromNativePaths.csproj" (целевой объект CopyNativeProjectBinaries ) (4506:2) ->
                     d:\net\coreclr\tests\src\dir.targets(112,6): error : The native project files are missing in d:\net\coreclr\tests\..\bin\tests\obj\Windows_NT.x64.Debug\src\Loader\NativeLibs\Debug\ please run build from the root of the repo at least once [d:\net\coreclr\tests\src\Loader\NativeLibs\FromNativePaths.csproj]

    Предупреждений: 7
    Ошибок: 4

Затраченное время: 00:09:22.78

```

### PR DESCRIPTION
**Pull request by ramarag**
This fixes #2217 

@gkhanna79 @schellap  @mmitche @AndyAyersMS  PTAL

### CODE REVIEW

**Comment by ramarag** [comment]
@dotnet-bot  test OSX x64 Release Build and Test
Ubuntu x64 Release Build and Test

**Comment by ramarag** [comment]
@dotnet-bot  test 
OSX x64 Release Build and Test
Ubuntu x64 Release Build and Test

**Comment by ramarag** [comment]
@mmitche : are these spurious failures ?

**Comment by mmitche** [comment]
@ramarag No, you messed around with the test directories and the run-test script cannot find everything it needs..

**Comment by mmitche** [comment]
@ramarag Please investigate failures before retrying.

File: tests/buildtest.cmd [hunk 1]
```diff
@@ -87,12 +87,12 @@ if %__verbosity%==detailed (
...
 set "__BinDir=%__RootBinDir%\Product\%__BuildOS%.%__BuildArch%.%__BuildType%"
-set "__TestBinDir=%__RootBinDir%\tests\%__BuildOS%.%__BuildArch%.%__BuildType%"
+set "__TestBinDir=%__RootBinDir%\tests\%__BuildOS%.%__BuildArch%.%__BuildType%\"
 :: We have different managed and native intermediate dirs because the managed bits will include
 :: the configuration information deeper in the intermediates path.
 :: These variables are used by the msbuild project files.
-set "__NativeTestIntermediatesDir=%__RootBinDir%\tests\obj\%__BuildOS%.%__BuildArch%.%__BuildType%"
-set "__ManagedTestIntermediatesDir=%__RootBinDir%\tests\obj"
+set "__NativeTestIntermediatesDir=%__RootBinDir%\tests\obj\Native\%__BuildOS%.%__BuildArch%.%__BuildType%"
+set "__ManagedTestIntermediatesDir=%__RootBinDir%\tests\obj\Managed"
```
Comment (gkhanna79) [inline tests/buildtest.cmd]: Is there a specific reason to not have managed intermediates go under a OS/Arch/Type subfolder like the native ones?

**Comment by gkhanna79** [comment]
Other than the question above, LGTM.

**Comment by schellap** [comment]
LGTM.

File: tests/buildtest.cmd (hunk 1 above)
Comment (ramarag) [inline tests/buildtest.cmd]: Fixed it

**Comment by gkhanna79** [comment]
@ramarag The corresponding issue is marked for RC2. Please take it through the RC2 process.

**Comment by ramarag** [comment]
already done :https://github.com/dotnet/coreclr/pull/2789

**Pull request merged**